src/platforms/desktop.c

```diff
@@ -69,6 +69,20 @@ void platform_init(void) {
         log_fatal("Error initializing SDL");
     }
 
+#ifdef _WIN32
+    #define PLATFORM_RENDER_HINT "direct3d"
+#else
+    #define PLATFORM_RENDER_HINT "opengl"
+#endif
+
+    // Use a platform-specific rendering hint to optimize performance and compatibility
+    // while addressing issues with blurry rendering on some systems, particularly high-DPI displays.
+    // This approach allows for:
+    // 1. Improved performance and pixel-perfect rendering on most systems
+    // 2. Better compatibility across different platforms
+    // 3. Flexibility to change the hint based on the target platform
+    SDL_SetHint(SDL_HINT_RENDER_DRIVER, PLATFORM_RENDER_HINT);

```

Comment:
I'm trying to understand why this is working for you. You are building on a Mac which doesn't support OpenGL? Why is this fixing your issue?

Reply:
Well, the thing is, no recent Macs support OpenGL – Starting from macOS 10.14 Mojave (released in 2018), Apple announced that OpenGL and OpenCL are deprecated (in favor of Metal).

Even though the hint is set to `opengl`, SDL2 is smart enough to choose an appropriate rendering backend for the platform. On modern Macs, it's likely using Metal behind the scenes, regardless of the OpenGL hint.

This hint doesn't force SDL to use OpenGL. It merely suggests which rendering backend to prefer. On macOS, even though the hint suggests `opengl`, SDL2 is likely choosing a more appropriate backend, probably Metal. SDL2 is designed to work across platforms and will select a working renderer even if the hinted one isn't available.

The real rendering work is done by SDL functions like `SDL_CreateRenderer()`, `SDL_RenderCopy()`, and `SDL_RenderPresent()`. These functions use whatever backend SDL has chosen, which on a modern Mac is likely Metal.

It's working because SDL2 is handling all the low-level graphics API interactions. The code doesn't need to use OpenGL directly; SDL2 takes care of selecting and using an appropriate graphics backend for the platform it's running on.

I don't know why exactly this hint made it work, but it's working. And it's a hint! It's not a rule for SDL to bound to. See, it's Metal in the inside.

Reply:
Could we just special case it for Mac and let the other platforms use default behavior? Thoughts?

```C
#ifdef __APPLE__
    SDL_SetHint(SDL_HINT_RENDER_DRIVER, "metal"); // Or opengl if metal is what is breaking you?
#endif
```